extensions/common/context.h

```diff
@@ -202,5 +212,22 @@ google::protobuf::util::Status extractNodeMetadataValue(
     const google::protobuf::Struct& node_metadata,
     google::protobuf::Struct* metadata);
 
+class FlatNodeCache {
+ public:
+  FlatNodeCache() {}
```

Comment:
nit: = default;
also, if updateSize is required it is probably better to set max_size in ctor. Otherwise, set max_size_ to a reasonable value to avoid indeterministic value.

Reply:
removed the super class altogether since it's only used once.